src/TAlias.cpp

```diff
@@ -323,35 +349,49 @@ void TAlias::compileAll()
     mNeedsToBeCompiled = true;
     if( ! compileScript() )
     {
-        if( mudlet::debugMode ) {TDebug(QColor(Qt::white),QColor(Qt::red))<<"ERROR: Lua compile error. compiling script of alias:"<<mName<<"\n">>0;}
+        if( mudlet::debugMode ) {
+            TDebug( QColor(Qt::white), QColor(Qt::red) ) << "LUA ERROR: when compiling script of alias:"
+                                                         << mName
+                                                         << "\n"
+                                                         >> 0;
+        }
         mOK_code = false;
     }
-    typedef list<TAlias *>::const_iterator I;
-    for( I it = mpMyChildrenList->begin(); it != mpMyChildrenList->end(); it++)
-    {
-        TAlias * pChild = *it;
-        pChild->compileAll();
-    }
-}
-
-void TAlias::compile()
-{
-    if( mNeedsToBeCompiled )
+    else
     {
-        if( ! compileScript() )
-        {
-            if( mudlet::debugMode ) {TDebug(QColor(Qt::white),QColor(Qt::red))<<"ERROR: Lua compile error. compiling script of alias:"<<mName<<"\n">>0;}
-            mOK_code = false;
-        }
+        TDebug( QColor(Qt::white), QColor(Qt::red) ) << "LUA OK: when compiling script of alias:"
+                                                     << mName
+                                                     << "\n"
+                                                     >> 0;
     }
+    compileRegex(); // Effectively will repost the error if there was a problem in the regex
     typedef list<TAlias *>::const_iterator I;
     for( I it = mpMyChildrenList->begin(); it != mpMyChildrenList->end(); it++)
     {
         TAlias * pChild = *it;
-        pChild->compile();
+        pChild->compileAll();
     }
 }
 
+// Not used - so commented out rather than updating error/success messages
```

Comment:
@vadi2 Can you clarify "Yes please no such shenanigans in release_30."? :confounded:

I have a commit ready to push out to here that:
* makes the `TAlias::mErrorMessage` contain an error message __including__ HTML formatting tags to make the error message __bold__ AND blue for regex compile messages - so that effect manifests when `dlgTriggerEditor::showError(...)` is called with it. {As requested}
* Change the description of the item with an error in the Central Debug Console to be the "...lua code of..." from "...script..." {As requested for `TAlias` class - however rather than leaving it just for THAT class I extended the change to ALL relevant classes for consistency}
* Whilst looking through the code for the above I identified the following additional unused methods {besides the existing `(void)TAlias::compile()` and `(void)TTrigger::compile()`} and (so far) have commented them out:
  * `(void)TAction::compile()`
  * `(void)TKey::compile()`
  * `(void)TLuaInterpreter::compile( QString & )`
  * `(void)TLuaInterpreter::compileScript( QString & )`
  * `(void)TTimer::compile()`
{Only `TScript` has a `compile()` that is used!}
* I realised it was easiest to re-factor the code to make the Central Debug Console messages for the `TXxxxx::compileAll()` methods be generated centrally in the `TLuaInterpreter::compile(...)`; I also spotted that the Central Debug Console often ends up showing a long list of identical messages to the error that some Lua code (for a Mudlet item) compiled OK (or not) which is a bit unhelpful from a logging point of view - I was able to revise the code as part of other editing for this PR and an (optional) identifying `QString` {defaults to empty `QString()`} has been added to `TLuaInterpreter::compile(...)` {the one that was NOT commented out :grinning:} which is inserted to revised "LUA OK:" or "LUA ERROR:" messages to say WHAT (in a couple of words plus the item name e.g. "...in {Trigger|Key|...etc} \"myItem\"...") was successfully or unsuccessfully compiled. This includes Lua code that is part of a "Lua condition" in a trigger (which can be picked out by the inclusion of an addition "item #", where # is the item number, as part of that identifier).

Whilst I think the second pair of points are worthwhile and constructive inclusions I am aware that this may count as __creature feep__ - or whatever the term is.  OTOH if I do not comment out or remove the unused code I might then have to edit it to fit into this revised paradigm for those Central Debug Console messages instead of just forgetting about it as I would like.

So, should I _publish_, err, push _and be damned_ or do I rethink it :question: